examples/bri-3/docs/zeroKnowledge/circuits/snarkjs.md

```diff
@@ -0,0 +1,57 @@
+# Snarkjs Circuit
+
+This document describes the setup and process for generating zero-knowledge proofs using snarkjs. The circuits are written and compiled using [Circom](https://docs.circom.io/getting-started/installation/). Then, the [snarkjs](https://github.com/iden3/snarkjs) library is used to generate proofs from these compiled circuits.
+
+> **Note**
+> 
+> There are other available zk circuit libraries such as gnark.
+> 
+> Here is a [link](https://blog.celer.network/2023/03/01/the-pantheon-of-zero-knowledge-proof-development-frameworks/) for comparing performances of gnark and snarkjs circuit libraries. To summarise,
+> - Proof generation time: For Groth16, gnark is 5~10 times faster than snarkjs
+> - Peak memory usage: It is comparable for both libraries.
+> - CPU utilisation: Gnark shows better CPU utilisation.
+> 
+> As discussed during call on 27/04/2023, snarkjs is preferred due to its ease of development (a separate service for running zk-circuits in golang is not required). Also, It is deemed sufficient for our use case based on the number of constraints in our circuit.
+> 
+> The zero knowledge proof module interface allows these libraries to be used interchangeably. According to your requirement, you may write new circuits in your preferred library and plug it into this module.
+
+## Pre-Setup Installations
+
+[Detailed Instructions](https://docs.circom.io/getting-started/installation/#installing-dependencies)
+
+Step 1: Install Rust
+
+Circom is written in Rust.
+
+`curl --proto '=https' --tlsv1.2 https://sh.rustup.rs -sSf | sh`
+
+Step 2: Install Circom
+
+Run the following commands from the directory where your computer stores global libraries.
+
+`git clone https://github.com/iden3/circom.git`
+
+`cd circom`
+
+`cargo build --release`
+
+`cargo install --path circom`
+
+Step 3: Install Snarkjs
+
+`npm install -g snarkjs`
```

Comment:
small thing, but could this be part of repo dependencies and not global installation, since we are running scripts with npm run (i am not sure about this tbh, but maybe worth checking, up to you)

Reply:
In future PRs, I will add scripts that use snarkjs cli commands. Installing it globally makes it convenient for the user to use these commands. If it is only part of the repo dependencies, then the user will need to manually add the library to his $PATH or call the commands using the full file path.